### DISCUSSION TITLE
Various Ideas about Textbox

### DISCUSSION THREAD

**Discussion by rRobis** [SubtitleEdit/subtitleedit#8254] [Ideas]
When fetching the text from next subtitle, it would be great to see couple words forward, without looking at the list view.

(i took image from issue that i made, but i guess it makes sense to only have the next subtitle ghost, instead of the previous too)
![image](https://github.com/SubtitleEdit/subtitleedit/assets/19517183/22af35d8-da19-43a6-b941-a948e061d1bd)

**Comment by LeonCheung** [comment]
Not sure if I could help you on this but here is my shortcut setting:

![image](https://github.com/SubtitleEdit/subtitleedit/assets/494788/d456fd12-3c86-45a9-82b9-b0a2b46f4c6b)

With a setting like the above, you can fetch (or reverse) several words per second. I think it's faster than judging by the naked eye.  :)

**Reply by rRobis** [reply]
Thanks for the tips. I have pretty much the same setup, but i've made with AHK so that i can do it by modifiers+scroll with mouse, instead of arrows

So the only remaining problem is that i don't know if i should scroll more or not, because i'm not sure what the next couple words will be.

**Comment by LeonCheung** [comment]
Haha, guess what, just come up with a cooler idea (a new feature, inspired by the word "scroll" you typed), which could be possibly implemented by dev.

We could fetch down / move up the first letter by holding **Shift + mouse wheel scrolling** up or down, and do the same for the end word by holding **Alt + mouse wheel scrolling** up or down. 😆 

That would be way faster and more intuitive. (Ctrl key should not be used, because it's used for scaling when syntax coloring is enabled)

But better to have it as an optional setting to enable/disable because some ppl may find it's a bit risky to mess up with things if they trigger such operations by chance.

**Comment by rRobis** [comment]
We should have shortcuts to move around **original subtitle text** (at least the first two, that fetch words from next subtitle), when we are in translator mode.

I know that it's possible to click on the original one and then it will work for it, but sometimes it makes the workflow a little bit annoying.

![image](https://github.com/SubtitleEdit/subtitleedit/assets/19517183/14c0589a-2834-4223-9b1a-893804433a30)

@niksedk i guess this one shouldn't be too hard to implement?

**Comment by rRobis** [comment]
@niksedk I have another idea about textbox.

Would it be possible, to make an option, to mirror the word break from video to text box? 

Let's say, if in video I have only 2 word fitting in the width, then textbox should also show same break in textbox even if it has a lot of text in it overflowing (same 2 words in the first line etc). It would help to work with textbox only, and not look at the video, which sometimes can be distracting. 

I guess it's probably not easy to do (except if there's some way to get a report from ffmpeg of how the text is broken up?)
